src/tools/fuzzing/heap-types.cpp

```diff
@@ -162,35 +172,47 @@ struct HeapTypeGeneratorImpl {
         .add(FeatureSet::SIMD, Type::v128));
   }
 
-  HeapType generateHeapType() {
+  HeapType generateHeapType(Shareability share) {
     if (rand.oneIn(4)) {
-      return generateBasicHeapType();
-    } else {
-      Index i = rand.upTo(recGroupEnds[index]);
-      return builder[i];
+      return generateBasicHeapType(share);
+    }
+    if (share == Shared) {
+      // We can only reference other shared types.
+      std::vector<Index> eligible;
+      for (Index i = 0, n = recGroupEnds[index]; i < n; ++i) {
```

Comment:
```suggestion
      for (Index i = 0; i < recGroupEnds[index]; ++i) {
```
That was an optimization, I assume? I trust the compiler to LICM it, so it seems better to keep the code shorter.

Reply:
I don't think the compiler will be able to prove that the contents of `recGroupEnds[index]` will not be modified by the loop body, and this pattern is not too rare, so I think this code is reasonable as-is.

Reply:
Fair enough. I don't feel it's a critical line to optimize but sgtm.